cmds/core/gpgv/gpgv.go

```diff
@@ -113,10 +83,45 @@ func verifyDetachedSignature(key *packet.PublicKey, contentf, sigf io.Reader) er
 	switch sig := p.(type) {
 	case *packet.Signature:
 		err = key.VerifySignature(h, sig)
-	case *packet.SignatureV3:
-		err = key.VerifySignatureV3(h, sig)
 	default:
-		panic("unreachable")
+		return fmt.Errorf("unknown signature")
 	}
 	return err
 }
+
+func runGPGV(w io.Writer, verbose bool, keyfile, sigfile, datafile string) error {
+	if keyfile == "" || sigfile == "" || datafile == "" {
+		return errUsage
+	}
+
+	keyf, err := os.Open(keyfile)
+	if err != nil {
+		return err
+	}
+	sigf, err := os.Open(sigfile)
+	if err != nil {
+		return err
+	}
+	contentf, err := os.Open(datafile)
+	if err != nil {
+		return err
+	}
+
+	key, err := readPublicSigningKey(keyf)
+	if err != nil {
+		return fmt.Errorf("key: %v ", err)
+	}
+
+	if err = verifyDetachedSignature(key, contentf, sigf); err != nil {
+		return fmt.Errorf("verify: %v", err)
+	}
+	fmt.Fprintf(w, "OK")
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	if err := runGPGV(os.Stdout, verbose, flag.Args()[1], flag.Args()[1], flag.Args()[3]); err != nil {
```

Comment:
This will segv the first time it's run with < 4 args I guess? Also, why is it flag.Args of 1, 1, and 3? Should it be flag.Args 0, 1, 2? os.Args is different, flag.Args filters the command name, for example.

Reply:
This by the way shows the danger of this approach: we're never testing main() it seems?

Reply:
Also shows the danger of lacking focus on my side. Good we have reviews :)